rest/config_test.go

```diff
@@ -37,16 +37,19 @@ func TestReadServerConfig(t *testing.T) {
 			name:   "unknown field",
 			config: `{"invalid": true}`,
 			err:    `json: unknown field "invalid": unrecognized config value`,
+			errEE:  `found unknown field: invalid`,
 		},
 		{
 			name:   "incorrect type",
 			config: `{"logging": true}`,
 			err:    `json: cannot unmarshal bool into Go struct field ServerConfig.Logging of type base.LoggingConfig`,
+			errEE:  `expect { or n, but found t`,
```

Comment:
The error messages are a bit less useful when identifying what might be going wrong, but I guess there isn't much to be done about that.

Reply:
The full error message returned is more helpful, in that it gives additional context as to where this error happened. But it is a little bit more obscure than the stdlib errors.

jsoniter errors:
```
TestReadServerConfig/unknown_field
rest.ServerConfig.ReadObject: found unknown field: invalid, error found in #10 byte of ...|{"invalid": true}|..., bigger context ...|{"invalid": true}|...: unrecognized config value

TestReadServerConfig/incorrect_type
rest.ServerConfig.Logging: readObjectStart: expect { or n, but found t, error found in #10 byte of ...|ogging": true}|..., bigger context ...|{"logging": true}|...

TestReadServerConfig/invalid_JSON
rest.ServerConfig.ReadString: expects " or n, but found t, error found in #2 byte of ...|{true}|..., bigger context ...|{true}|...
```

stdlib errors:
```
TestReadServerConfig/unknown_field
json: unknown field "invalid": unrecognized config value

TestReadServerConfig/incorrect_type
json: cannot unmarshal bool into Go struct field ServerConfig.Logging of type base.LoggingConfig

TestReadServerConfig/invalid_JSON
invalid character 't' looking for beginning of object key string
```